TestScrapyProject/e_commerce/spiders/selected.py

```diff
@@ -0,0 +1,156 @@
+import re
+import json
+from copy import deepcopy
+from urllib.parse import urljoin
+
+from scrapy import Request, Spider
+
+from ..items import ProductItem, SizeItem
+
+
+class SelectedSpider(Spider):
+    name = 'selected'
+    PAGINATION_API = 'https://www.selected.com/on/demandware.store/Sites-sl-Site/en_GB/Search-UpdateGrid?cgid='
+    countries_info = [
+        ('gb', 'GBP', 'en', "https://www.selected.com/en-gb/home")]
+
+    PAGE_TEXT_PATTERN = re.compile(r'\d+')
+
+    def start_requests(self):
+        for country_code, currency, language, country_url in self.countries_info:
+            meta = {
+                'language_code': language,
+                'currency': currency,
+                'country_code': country_code,
+            }
+            yield Request(country_url, self.parse_top_categories, meta=meta)
+
+    def parse_top_categories(self, response):
+        for level1 in response.xpath('//div[contains(@class, "menu-burger__category--root")]/p'):
+            level1_subcat = self.parse_subcat(response, level1)
+            yield from self.make_navigation_request(response, [level1])
+
+            for level2 in level1_subcat.xpath('p[contains(@class, "category-level-2")]'):
+                level2_subcat = self.parse_subcat(response, level2)
+                yield from self.make_navigation_request(response, [level1, level2])
+
+                for level3 in level2_subcat.xpath('p[contains(@class, "category-level-3")]'):
+                    yield from self.make_navigation_request(response, [level1, level2, level3])
+
+    def make_navigation_request(self, response, selectors):
+        cat_url = selectors[-1].css('a::attr(href)').get()
+        if not cat_url:
+            return
+
+        meta = response.meta.copy()
+        categories = [sel.css('a::text').get().strip() for sel in selectors]
+        cat_id = selectors[-1].attrib.get('data-category-item-id', '')
+        meta.update({'categories': categories,
+                     'cat_url': urljoin(
+                         response.url, cat_url), 'cat_id': cat_id
+                     })
+        self.total_pages = 0
```

Comment:
extract total page in make_pagination function

parse_pagination is a better name then make_pagination